cpp/src/arrow/compute/kernels/sort_to_indices.cc

```diff
@@ -139,6 +150,54 @@ class CountSorter {
   }
 };
 
+template <typename ArrowType, typename Comparator>
+class CountCompareSorter {
```

Comment:
"CountOrCompareSorter"? Also, please add a comment or docstring explaining what this does.

Reply:
Done